packages/dd-trace/src/priority_sampler.js

```diff
@@ -87,6 +98,48 @@ class PrioritySampler {
       }
     }
   }
+
+  _isSampledByRule (context, rule) {
+    const sampled = this._limiter.isAllowed() && rule.sampler.isSampled(context)
+
+    context._metrics[SAMPLING_LIMIT_DECISION] = this._limiter.effectiveRate()
+    context._metrics[SAMPLING_RULE_DECISION] = rule.sampleRate
```

Comment:
we only want to set these if we used that method for sampling.

e.g.

```js
context._metrics[SAMPLING_RULE_DECISION] = rule.sampleRate
if (!rule.sampler.isSampled(context)) return false

const sampled = this._limiter.isAllowed()
context._metrics[SAMPLING_LIMIT_DECISION] = this._limiter.effectiveRate()
return sampled
```

(ish)

Reply:
The reason for that is so we can debug why a span was sampled or not. e.g. if we see `SAMPLING_RULE_DECISION` only, then we can assume it was a p0 because of the rule, not because of rate limiter.